org.sonarlint.eclipse.cdt/src/org/sonarlint/eclipse/cdt/internal/CProjectConfiguratorExtension.java

```diff
@@ -63,17 +64,9 @@ private static boolean isCdtPresent() {
   }
 
   @Override
-  public Set<String> whitelistedPlugins() {
-    if (isCdtPresent()) {
-      return Collections.singleton("cpp");
-    }
-    return Collections.emptySet();
-  }
-
-  @Override
-  public Set<Language> whitelistedLanguages() {
+  public Set<SonarLintLanguage> enableLanguages() {
     // Objective-C is not supported by CDT!
-    return isCdtPresent() ? SonarLintUtils.CONNECTED_MODE_LANGUAGES_CDT : Collections.emptySet();
+    return isCdtPresent() ? EnumSet.of(SonarLintLanguage.C, SonarLintLanguage.CPP) : Collections.emptySet();
```

Comment:
Why is it not referring to `SonarLintUtils.OPTIONAL_CONNECTED_LANGUAGES`?

Reply:
Because in the future, `SonarLintUtils.OPTIONAL_CONNECTED_LANGUAGES` can grow.